capstone-backend/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -45,29 +45,42 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    // response.setHeader("Access-Control-Allow-Methods", "GET");
-    // response.setHeader("Access-Control-Allow-Credentials", "true");
-    // response.setHeader("Access-Control-Allow-Origin",
-    //     "https://3000-bbaec244-5a54-4467-aed6-91c386e88c1a.ws-us02.gitpod.io");
-
     String fullOutput = "";
-    String searchTerm = request.getParameter("searchTerm");
-    if (searchTerm == null || searchTerm.isEmpty())
-      return;
-    searchTerm = URLEncoder.encode(searchTerm, "UTF-8");
+    String formattedURL = "";
 
-    int maxResults = 0;
-    String maxResultsParam = request.getParameter("maxResults"); 
-    if (maxResultsParam != null) { 
-      maxResults = parseNaturalNumber(maxResultsParam);
-    }
-    if (maxResultsParam == null || maxResults == -1) {
-      maxResults = DEFAULT_MAX_RESULTS;
+    if (request.getParameter("gbookId") != null) {
+      if (request.getParameterMap().size() > 1) {
+        System.err.println("Error: No other parameter can be sent with a gbookId");
+        response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      // make formatted url
+      String gbookId = request.getParameter("gbookId");
+      formattedURL = String.format("https://www.googleapis.com/books/v1/volumes/%s", gbookId);
+
+    } else if (request.getParameter("searchTerm") != null || request.getParameter("searchTerm").isEmpty()) {
```

Comment:
Just a quick clarification on this condition - I see that in your previous implementation, you're returning if either the `searchTerm` is `null` or `isEmpty`. Therefore, should we have this condition be the "negation" of that previous condition, i.e.,

`else if (request.getParameter("searchTerm") != null && !request.getParameter("searchTerm").isEmpty())`

Reply:
Yes, this is correct! I just missed the ! in front of the second part of that statement

Thank you for catching this!